hack/generate/samples/ansible/memcached.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ansible
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/operator-framework/operator-sdk/hack/generate/samples/pkg"
+	"github.com/operator-framework/operator-sdk/test/utils"
+	testutils "github.com/operator-framework/operator-sdk/test/utils"
+)
+
+// MemcachedAnsible defines the context for the sample
+type MemcachedAnsible struct {
+	ctx *pkg.SampleContext
+}
+
+// NewMemcachedAnsible return a MemcachedAnsible
+func NewMemcachedAnsible(ctx *pkg.SampleContext) MemcachedAnsible {
+	return MemcachedAnsible{ctx}
+}
+
+// Prepare prepare the directory for the Memcached Ansible Sample
+func (ma *MemcachedAnsible) Prepare() {
+	log.Infof("destroying directory for memcached Ansible samples")
+	ma.ctx.Destroy()
+
+	log.Infof("creating directory for Ansible Sample")
+	err := ma.ctx.Prepare()
+	pkg.CheckError("error to creating directory for Ansible Sample", err)
+
+	log.Infof("setting domain and GKV")
+	ma.ctx.Domain = "example.com"
+	ma.ctx.Version = "v1alpha1"
+	ma.ctx.Group = "cache"
+	ma.ctx.Kind = "Memcached"
+}
+
+// Run the steps to create the Memcached Ansible Sample
+func (ma *MemcachedAnsible) Run() {
+	log.Infof("creating the project")
+	err := ma.ctx.Init(
+		"--plugins", "ansible",
+		"--group", ma.ctx.Group,
+		"--version", ma.ctx.Version,
+		"--kind", ma.ctx.Kind,
+		"--domain", ma.ctx.Domain,
+		"--generate-role")
+	pkg.CheckError("creating the project", err)
+
+	err = ma.ctx.Make("kustomize")
+	pkg.CheckError("error to scaffold api", err)
+
+	log.Infof("customizing the sample")
+	err = utils.UncommentCode(
+		filepath.Join(ma.ctx.Dir, "config", "default", "kustomization.yaml"),
+		"#- ../prometheus", "#")
+	pkg.CheckError("enabling prometheus metrics", err)
+
+	ma.addingAnsibleTask()
+	ma.addingMoleculeMockData()
+
+	pkg.RunOlmIntegration(ma.ctx)
+}
+
+// addingMoleculeMockData will customize the molecule data
+func (ma *MemcachedAnsible) addingMoleculeMockData() {
+	log.Infof("adding molecule test for Ansible task")
+	moleculeTaskPath := filepath.Join(ma.ctx.Dir, "molecule", "default", "tasks",
+		fmt.Sprintf("%s_test.yml", strings.ToLower(ma.ctx.Kind)))
+
+	err := utils.ReplaceInFile(moleculeTaskPath,
+		"false", "true")
+	pkg.CheckError("replacing molecule default tasks", err)
```

Comment:
In the samples repo, I replaced this value with true so we could leave the scaffolded test but make it pass, adding a new file for the new tests. Since you are doing this all in a single file, I don't think it is valuable to edit it in place, why not just template the whole task?

Reply:
Catcher. Yep, it is true. Good spot.